python_modules/libraries/dagster-databricks/dagster_databricks/databricks.py

```diff
@@ -38,9 +39,15 @@ def __init__(self, host: str, token: str, workspace_id: Optional[str] = None):
         # confusing for users to use since this is an unofficial wrapper around the documented
         # Databricks REST API. We should consider removing this in the future.
         self.client = DatabricksAPI(host=host, token=token)
+        self.__setup_user_agent(self.client.client)
 
         # Expose an interface directly to the official Databricks API client.
         self._api_client = ApiClient(host=host, token=token)
+        self.__setup_user_agent(self._api_client)
+
+    def __setup_user_agent(self, client: ApiClient) -> None:
```

Comment:
why not just pass in the default headers when creating the DattabricksAPI / ApiClient objects?

Reply:
It will be overridden by the client during initialization, so we have to wait until afterwards to set these values.

https://github.com/databricks/databricks-cli/blob/fd2ad70d1cbe03b587970a8f41d17004d453707c/databricks_cli/sdk/api_client.py#L126-L131